engine/runtime-parser/src/main/scala/org/enso/compiler/core/ir/expression/errors/Redefined.scala

```diff
@@ -195,7 +195,7 @@ object Redefined {
 
     /** @inheritdoc */
     override def message(source: Source): String =
-      s"Method overloads are not supported: ${targetType.map(_.name + ".").getOrElse("")}from " +
+      s"Ambiguous conversion: ${targetType.map(_.name + ".").getOrElse("")}from " +
```

Comment:
I think that this error message may be clearer - conversions work a bit differently than method overloads - they **do overload** the `from` method, just with different types of `that`.

Reply:
What if I accidentaly do something like this:
```
method x y = x + y
method x = x
```

Would the error message say "Ambiguous conversion" in that case as well? Isn't that misleading? What about trying to distinguish between those two cases - "Ambiguous conversion" VS "Method overload"?

Reply:
We _do_ distinguish these cases already.

The error message I have edited was only relevant for conversions, method overloads have a separate error class - I assume the error message for conversions was copy-pasted from overloads and that's why it used to say the same thing.

Reply:
You can see [here](https://github.com/enso-org/enso/blob/b9328bd03d7bda5650c27788682b0370bcb3db4d/engine/runtime/src/test/scala/org/enso/interpreter/test/semantic/OverloadsResolutionErrorTest.scala#L40) a test verifying that the behaviour is exactly as you'd expect it to be.

Reply:
Well, the above test was for member methods - so just to be comprehensive I added a test case like the one you mentioned:
 [2a37e54](https://github.com/enso-org/enso/pull/8245/commits/2a37e540383c9753e6c3bf2c42b1feb5ad43427f)